dev-lang/rust_bin/rust_bin-1.29.0.recipe

```diff
@@ -0,0 +1,101 @@
+SUMMARY="Modern and safe systems programming language"
+DESCRIPTION="Rust is a systems programming language that runs blazingly fast, \
+prevents almost all crashes*, and eliminates data races."
+HOMEPAGE="https://www.rust-lang.org/"
+COPYRIGHT="2018 The Rust Project Developers"
+LICENSE="MIT"
+REVISION="1"
+
+case "$effectiveTargetArchitecture" in
+	x86)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rust-$portVersion-i686-unknown-haiku.tar.xz"
+CHECKSUM_SHA256="b4d40d82bb6c07746b1ca8e407a23e45dd357744ee80217446bc9f25cc5e9eaa"
+SOURCE_DIR="rust-$portVersion-i686-unknown-haiku"
+		;;
+	x86_64)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rust-$portVersion-x86_64-unknown-haiku.tar.xz"
+CHECKSUM_SHA256="e8229ee38453978485720e6a8ef84e0474aa5fb9375b3e74ecce655ee742b5f2"
+SOURCE_DIR="rust-$portVersion-x86_64-unknown-haiku"
+		;;
+	*)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rustc-$portVersion-src.tar.xz"
```

Comment:
What's the point of having the rust source in a recipe for binaries?

Reply:
It was suggested in the discussion about the  recipe for Rust 1.21.0. 
See this comment: https://github.com/haikuports/haikuports/pull/2647#issuecomment-399678783

Reply:
The only point might be to have a correct recipe in case effectiveTargetArchitecture isn't set.

Reply:
I have no opinion on the existence of this line. If you and @fbrosson agree on taking it out I will take it out.